### PR TITLE
Add Consistent Hash

### PR DESCRIPTION
**Pull request by gogit**
Add Consistent Hash and File Lookup

### CODE REVIEW

File: file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java [hunk 1]
```diff
@@ -1,20 +1,43 @@
...
         final String fileName = fl.getFileName(uuid);
-        Assert.assertTrue(fileName, fl.getFileName(uuid).matches("/gluster-storage/dir-[0-9]"));
+        assertTrue(fileName, fileName.matches(
+                        String.join("","/gluster-storage/dir-[0-9]/", uuid.toString())));
     }
+    
+    @Test
+    public void shouldTestConsistentLookupFileName() {
+        final DirectoryPath dp = new DirectoryPath();
+        final FileLookup fl = new FileLookup(dp);
```
Comment (mapingo) [inline file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java]: use variables names that describe, for example fileLookup, would be better.  dp is used only once, and can be inlined

File: file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java [hunk 2]
```diff
@@ -1,20 +1,43 @@
...
+        final Map<UUID, String> fileNames = new TreeMap<>();
+        final List<UUID> l = Stream.generate(UUID::randomUUID).limit(100000)
```
Comment (mapingo) [inline file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java]: As above full descriptive name required

File: file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java [hunk 3]
```diff
@@ -1,20 +1,43 @@
...
+                        .collect(Collectors.toList());
+
+        l.forEach(x -> {
```
Comment (mapingo) [inline file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java]: although x is commonly used, I think uuid would be better here

File: file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java [hunk 4]
```diff
@@ -1,20 +1,43 @@
...
+            fileNames.put(x, fl.getFileName(x));
+        });
 
-}
+        for(Map.Entry<UUID, String> me: fileNames.entrySet()){
```
Comment (mapingo) [inline file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java]: as above me could be more descriptive

**Review by mapingo — CHANGES_REQUESTED** [review]

File: file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java (hunk 1 above)
Comment (gogit) [reply file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java]: done

File: file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java (hunk 2 above)
Comment (gogit) [reply file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java]: done

File: file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java (hunk 3 above)
Comment (gogit) [reply file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java]: done

File: file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java (hunk 4 above)
Comment (gogit) [reply file-service-core/src/test/java/uk/gov/justice/services/fileservice/FileLookupTest.java]: done

**Review by mapingo — APPROVED** [review]

**Pull request merged**